heareval/tasks/util/luigi.py

```diff
@@ -63,6 +81,56 @@ def stage_number(self) -> int:
             raise ValueError("Unknown requires: %s" % self.requires())
 
 
+class DownloadCorpus(WorkTask):
+    """
+    Task for downloading a dataset to a specific filename
+    """
+
+    # URL to download dataset from
+    url = luigi.Parameter()
+    outfile = luigi.Parameter()
+
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def run(self):
+        download_file(self.url, os.path.join(self.workdir, self.outfile))
+        with self.output().open("w") as _:
+            pass
+
+    @property
+    def stage_number(self) -> int:
+        return 0
+
+
+class ExtractZipFile(WorkTask):
+    
+    infile = luigi.Parameter()
+    
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def requires(self):
+        #Some Task should be returned having a working directory from 
+        #which the infile will be picked.
+        raise NotImplementedError
```

Comment:
I don't love enforcing that this class needs to be overridden. Unless I'm missing something about how Luigi works under the hood (i.e. the requires must be set prior to instantiation for the task_id or something) -- I'm a bit more in favour of being about to just use these out of the box and set the requires using a lambda. Seems more efficient to me.